plugins/scanners/scancode/src/main/kotlin/ScanCodeResultModelMapper.kt

```diff
@@ -157,18 +157,15 @@ private fun getSpdxId(spdxLicenseKey: String?, key: String): String {
 /**
  * Map scan errors for all files using messages that contain the relative file path.
  */
-private fun mapScanErrors(result: ScanCodeResult): List<Issue> {
-    val input = result.headers.single().options.input.single()
-    return result.files.flatMap { file ->
-        val path = file.path.removePrefix(input).removePrefix("/")
+private fun mapScanErrors(result: ScanCodeResult): List<Issue> =
```

Comment:
> I don't think that's correct. The code just guards against the case if --strip-root was used.

The part of the commit message does not intent to talk about code within `mapScanErrors`, but about the code which maps license and copyright findings. That code indeed gets the paths wrong in case `--strip-root` has not been provided. Given that, I don't think it's wrong to say that the model mapper assumes `--strip-root`.

> As the command line to run ScanCode with can be configured by the user, the parser should work in any case, 

I agree it should work in the general case. But that would be an enhancement which requires the addition of further logic. The scope of this PR is only fixing a bug with the existing implementation / clarifying limitations.

> However, I agree that the prefix removal could be dropped here

Why only "could". It is inconsistent to drop the prefix in the error message, but not in license and copyright findings. This to me is  bug.

Reply:
I still believe the commit messages and code comments need rework to get the history facts right. When I rewrote much of the parser logic a while ago, the *intention* was to make the code work with or without `--strip-root` already back then, so IMO it does not make sense to document any requirement for `--strip-root` now.

So the line

    val path = file.path.removePrefix(input).removePrefix("/")

was supposed to strip a prefix *if* it's there, and use the path as-is otherwise. *That* is the bug. Instead of fixing the bug by adding code, we can simply remove the code (like you did) as the paths are made relative upstream. That should be made more clear, IMO.

I'd propose to simply have a single commit whose diff matches your second commit but says:

> If the `input` is just "." to denote the current directory and the `--strip-root` option was passed to ScanCode, then a leading dot is removed from "hidden" files, which is wrong. As paths are made relative via a scanner-agnostic post-processing of the scan summary anyway, simply remove this code to fix the bug.